chia/cmds/wallet_funcs.py

```diff
@@ -19,15 +18,31 @@
 from chia.wallet.util.wallet_types import WalletType
 
 
+def exp_to_dec(f):
+    float_string = repr(f)
```

Comment:
Switching to a float and then a string doesn't seem like an obviously good way to format the output from an arbitrary precision number such as a `Decimal`.  Why is this needed?  I feel like we ought to be able to either finder a suitable format specifier or otherwise directly process the decimal.

Reply:
One would think, but it's a glaring omission in the upstream format specifiers. The method here is more to detect when the default library outputs in exponential format and address only those instances. The conversion from there is simply done to keep code proxmitity/simplicty.  Manipulating directly would be "cleaner" but would essentially be replicating the core library functionality in a different way and at that point, one may as well focus on adding the feature upstream 🤷 

TL;DR: Doing direct would add unnecessary complication and obfuscation for not any tangible benefit that I can tell.  However, if I am totally off base and you know of a simpler way to accomplish it, I will gladly and humbly bow out.

Reply:
...for clarity: manipulating directly would require adding all extra core library logic to determine decimal point placement so that the appropriate string literal could be used for the output specifier.  Doing it in this way essentially lets the existing library do all that work, then we output from that.

Reply:
What specific values are problematic?

Reply:
I see the 1e-6 example in the OP, sorry.  I'll take a look around but it seems like just a `f` format specifier handles it down to 1e-12 anyways.  But, I haven't recreated the issue with 1e-6 in the repl yet, so we'll see.

Reply:
If I go to 1e-9 I can get the issue, but the `f` format handles it.

```console
$ python3.9 -c 'import decimal; amount = decimal.Decimal(1) / 1_000_000_000; print(str(amount), f"{amount:f}")'
1E-9 0.000000001
$ python3.9 -c 'import decimal; amount = decimal.Decimal(1) / 1_000_000_000_000; print(str(amount), f"{amount:f}")'
1E-12 0.000000000001
```

I'll have to try with my actual wallet later

Reply:
I am not a genius here and merely lifted the solution from elsewhere because I needed this feature myself.  But in following the many conversation around, the solution seemed reasonably simple since alternative approaches leave one needing to manage the precision.  I you want to make assumptions on precision, then there are certainly simpler approaches:

https://stackoverflow.com/questions/38847690/convert-float-to-string-in-positional-format-without-scientific-notation-and-fa

Reply:
Between using `decimal.Decimal` and having `divmod()` I think we could split it on our own, presuming we need to avoid stdlib formatting which certainly does have some not-preferred corners for some use cases.

This code isn't "optimized" but I think it is a bit more direct.  Given that the amount is already in mojo, we could skip the existing conversion to chia and this formatting would get an operation or two simpler.  Though, that's not a big deal for CLI output here probably.

<details>
<summary>Expand for example code and output</summary>

```python
import decimal
import math


mojo_per_chia = decimal.Decimal(1_000_000_000_000)
decimal_places = 12  # hard-coded, just to be super sure it is constant


def chia_to_string(chia: decimal.Decimal):
    just_chia, mojo_as_chia = divmod(chia, 1)
    just_mojo = (mojo_as_chia * mojo_per_chia).to_integral_exact()

    if just_mojo == 0:
        return str(just_chia)

    raw = f"{just_chia}.{just_mojo:0{decimal_places}}"
    return raw.rstrip("0")


examples = [
    decimal.Decimal("0.000_000_000_001"),
    decimal.Decimal("0.000_001"),
    decimal.Decimal("0.1"),
    decimal.Decimal("0"),
    decimal.Decimal("1"),
    decimal.Decimal("1.0"),
    decimal.Decimal("10"),
    decimal.Decimal("10.0"),
    decimal.Decimal("1.000_000_000_001"),
    decimal.Decimal("1_000_000_000_001"),
]


for example in examples:
    print()
    print(" str:", str(example))
    print("   f:", f"{example:f}")
    print("func:", chia_to_string(example))
```

```
 str: 1E-12
   f: 0.000000000001
func: 0.000000000001

 str: 0.000001
   f: 0.000001
func: 0.000001

 str: 0.1
   f: 0.1
func: 0.1

 str: 0
   f: 0
func: 0

 str: 1
   f: 1
func: 1

 str: 1.0
   f: 1.0
func: 1

 str: 10
   f: 10
func: 10

 str: 10.0
   f: 10.0
func: 10

 str: 1.000000000001
   f: 1.000000000001
func: 1.000000000001

 str: 1000000000001
   f: 1000000000001
func: 1000000000001
```

</details>

For the sake of compact PRs, further formatting should probably be handled separately, but at least an option to output with grouping separators on both sides of the decimal place would be nice.  Something like `1,037.190_949_000_001`, just to show the concept.  I don't know if there are any standards for this.